lib/rubycritic/source_control_systems/git.rb

```diff
@@ -52,6 +52,18 @@ def stashes_count
       def travel_to_original_state
         `git stash pop`
       end
+
+      def path_revisions_counts
+        `git log --name-only --format='' | sort | uniq -c`
```

Comment:
Interesting, I think it makes sense!

I just tried the `git log` command you suggested in a big repository and got these two warnings:

> warning: inexact rename detection was skipped due to too many files.
> warning: you may want to set your diff.renameLimit variable to at least 2098 and retry the command.

Not sure if we want to:

1. Suppress these warnings.
2. Fix the underlying cause of these warnings (I found [this](https://github.com/grimoirelab/perceval/issues/19) but not sure if it's an optimal solution).
3. Do nothing and leave it like this.

Don't let this discourage you though. If it's not a big effort try your suggested approach. I'd appreciate it if you could then compare your previous `churn` scores with the new scores.